util/buildRelease/gen_release

```diff
@@ -178,122 +102,259 @@ if (exists($ENV{"CHPL_GEN_RELEASE_NO_CLONE"})) {
-  SystemOrDie("make man");
-  SystemOrDie("make man-chpldoc");
-} else {
-  SystemOrDie("mkdir -p man/man1");
-}
-
-SystemOrDie("make clobber");
-
-print "[gen_release] Creating the examples directory...\n";
-SystemOrDie("rm examples"); # remove examples symbolic link
-SystemOrDie("cp -r test/release/examples .");
-SystemOrDie("cd util && cp start_test ../examples/");
-
-# print "[gen_release] Removing Makefiles that are not intended for release...\n";
-
-# print "[gen_release] Removing compiler directories that are not intended for release...\n";
-
-print "[gen_release] Removing runtime directories that are not ready for release...\n";
-SystemOrDie("cd runtime/src/launch && rm -r dummy");
-SystemOrDie("cd runtime/src/launch && rm -r mpirun");
-
-print "[gen_release] Removing frontend test directory which is not intended for release...\n";
-SystemOrDie("cd frontend && rm -r test");
-
-print "[gen_release] Removing third-party directories that are not intended for release...\n";
-SystemOrDie("cd third-party && rm *.devel*");
-SystemOrDie("cd third-party/chpl-venv && rm *.devel*");
-SystemOrDie("cd third-party/chpl-venv && rm chplspell-requirements.txt");
-
-print "[gen_release] Removing Git metadata files not intended for release...\n";
-SystemOrDie('find . -type f \( -name .gitignore -o -name .gitattributes \) -exec rm -f {} \; -print');
-
-chdir "$archive_dir";
-
-print "[gen_release] Chmodding the hierarchy\n";
-SystemOrDie("chmod -R ugo+rX .");
-SystemOrDie("chmod -R go-w .");
-
-foreach $file (@files) {
-    if (!(-e $file)) {
-        print "[gen_release] $file does not exist\n";
-        exit( 9);
-    }
-    push @tarfiles, "$reldir/$file";
-}
-
-foreach $dir (@code_dirs) {
-    @filelist = `find $dir`;
-    foreach $fullpath (@filelist) {
-        chomp $fullpath;
-        $file = $fullpath;
-        $file =~ s/(\S+\/)+//g;
-        if ($file =~ /(\.(h|cpp|c|ypp|lex)$)|Makefile|README|BUILD_VERSION|CMakeLists\.txt/) {
-            # print "$fullpath\n";
-            push @tarfiles, "$reldir/$fullpath";
-        }
-    }
-}
-
-foreach $dir (@complete_dirs) {
-    if (!(-e $dir)) {
-        print "[gen_release] $dir does not exist\n";
-    }
-    push @tarfiles, "$reldir/$dir";
-}
-
-
-if (! -d $resultdir) {
-    print("Creating $resultdir\n");
-    mkpath($resultdir, 1);
-}
-
-$tarball_name = "$resultdir/$reldir.tar.gz";
-$cmd = "tar -cz -f $tarball_name @tarfiles";
-chdir "..";
-print "[gen_release] $cmd\n";
-SystemOrDie ($cmd);
-
-print "[gen_release] Left result in $tarball_name\n";
-
-chdir $origCwd;
+    "tools/chpl-language-server",
+]
+
+
+def main():
+    version = ""
+    if len(sys.argv) > 1:
+        version = sys.argv[1]
+
+    reldir = f"chapel-{version}" if version else "chapel"
+
+    # Find CHPL_HOME. If unset, use root of repo relative to this script.
+    chplhome = os.getenv(
+        "CHPL_HOME",
+        os.path.dirname(
+            os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
```

Comment:
this entire block can become:

```Python
os.path.normpath(os.path.join(os.path.abspath(__file__), "../../.."))
```

I think it will even fit on one line.